### PR TITLE
172 Landing Pages

### LINKED ISSUE #172: As a user who is not yet logged in, I see a landing page based on the design
Description:
We have a [figma design](https://www.figma.com/file/irgQPLTrajxCXNiYBTEnMV/Login-%2F-Admin-Mockups-For-Feedback?node-id=0%3A1) for what a user sees when the user has not yet logged in. This story is to build out that design. This is a frontend only task. 

**What will not be done in this issue**
1. Updating the landing page with demo banner #235

AC: 

- [ ] When the user is not logged in, they see a landing page that matches the figma design
- [ ] This page renders at all form factors specified in the design (aka is "responsive")
- [ ] Use the text and placeholders from the mockup

Tasks:
- [ ] Create unauthenticated login.page
- [ ] Code documentation and annotation of why system is the way it is

DoD: 
 - [ ] minimum 90% code coverage test
 - [ ] 0 linting errors and warnings
 - [ ] 0 accessibility errors
 - [ ] deployed to a dev environment
 - [ ] all significant changes have been documented
 - [ ] free of high and medium security vulnerabilities
 - [ ] @lauraGgit has reviewed and signed off
 - [ ] @lfrohlich  has reviewed and signed off
 

Supporting Documentation:

(please include any relevant log snippets/ files/ screen shots here.)

Open Questions:

(any clarifying questions in support of the issue)

Resources: 
- https://www.figma.com/file/irgQPLTrajxCXNiYBTEnMV/Login-%2F-Admin-Mockups-For-Feedback?node-id=0%3A1

### PR DESCRIPTION
**Pull request by adamcaron**
### This pull request changes...

closes #172 

### Tasks: 

- [x] unauthenticated page is Welcome page
- [x] gov banner is left aligned, and blue background
- [x] responsive header
- [x] responsive jumbotron
- [x] sign in button works with the authentication flow
- [x] responsive footer
- [x] responsive main area
- [ ] Ensure alert bar is contained in a way that makes sense

### [AC] This pull request is ready to merge when...

- [x] Tests have been updated (and all tests are passing)
- [x] Vulnerability Scan Test (Zap) has no high/critcial CVEs
- [x] Linting Tests (ESLint for UI and Flake8 for Python)
- [x] Unit Tests ( including a code coverage test report)
- [x] Accessibility Tests (Pa11y UI Only)
- [ ] This code has been reviewed by someone other than the original author
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented
- [ ] The change has been documented
- [ ] Associated OpenAPI documentation has been updated
- [ ] Changelog is updated as appropriate

### [DoD] This feature is done when...

- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

**Comment by adamcaron** [comment]
### Fix 

ZAP scanner had been failing for awhile, throwing SASS errors.

The issue was at build time -- while running the OWASP Zap Scanner Vulnerability Test. We have a multistage Dockerfile. The first stage had access to a `NODE_SASS` variable via a `.env` file however the second stage did not have access to this -- and when the second stage ran `yarn build` the sass errors were being produced. An env variable was added to the build script in package.json:
```
{
  ...
  "scripts":
    ...
    "build": "SASS_PATH=node_modules:src react-scripts build",
  }
}
```
This makes it so whenever the app is built, the proper config is set.

It's not preferable to add it here from a documentation perspective because .json files don't allow for comments. It would have been nice to set the variable with an `ENV` layer in the Dockerfile and add explanatory comments, but that would only allow `yarn build` to succeed when run in Docker, and attempting to build the app outside of Docker would result in the error.

Unless we decided we _never_ want to run `yarn build` outside of the Docker image, this solution is probably best. Otherwise a future dev might try to build locally outside docker and it would fail, without much helpful sign as to why.

For now, the variable's purposes is documented in the frontend README under `yarn build` and also in the `.env.development` file, which sets the variable for the first Dockerfile stage in a local development environment.

**Comment by RafterGit** [comment]
@hilvitzs In addition to the Pa11y checks, this PR should address the following accessibility issues as well:

- Inline with [WCAG 1.3.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-programmatic.html) lets add a `visually-hidden` `span` class for `Sign in with Login.gov ` that says `Opens in a new website`. Something like this: `<button type="button" class="usa-button usa-button--big" data-testid="button">Sign in with Login.gov <span class="visually hidden">Opens in a new website</span></button>`
- Inline with [WCAG 1.1.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/text-equiv-all.html) add an `alt` text for ACF image in the footer
- Inline with [WCAG 2.4.3](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-order.html) After the user signs out and is sent back to the homepage, the focus should shift to the first `h1` tag (i.e. "Welcome") using `tabindex="-1"`. This can be done by Javascript

File: tdrs-frontend/src/components/Welcome/Welcome.jsx [hunk 1]
```diff
@@ -40,25 +30,64 @@ function Welcome() {
...
+              className="usa-media-block__img"
+              src={circleImg}
+              style={{ height: '120px' }}
+              alt="Alt text"
+            />
+            <img
+              className="usa-media-block__img"
+              src={circleImg}
+              style={{ height: '120px' }}
+              alt="Alt text"
```
Comment (RafterGit) [inline tdrs-frontend/src/components/Welcome/Welcome.jsx:84]: same comment as above for all "Alt text" references.

**Review by RafterGit — CHANGES_REQUESTED** [review]

File: tdrs-frontend/.env.development [hunk 2]
```diff
@@ -0,0 +1,11 @@
+REACT_APP_BACKEND_URL=http://localhost:8080/v1
+
+# Setup SCSS:
+# The following makes it possible to import SASS modules
+# without relative paths. Removing this will require all @imports in
+# .scss files to be make with relative paths, like '../../node_modules/my_package'
```
[the remark this answers is not in the document]
Comment (hilvitzs) [reply tdrs-frontend/.env.development]: fixed

File: tdrs-frontend/src/components/Header/Header.jsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+function HeaderComp() {
+  const [mobileNavOpen, setMobileNavOpen] = useState(false)
+
+  const toggleMobileNav = () => {
+    setMobileNavOpen((prevOpen) => !prevOpen)
+  }
+
+  const testItemsMenu = [
+    <a href="/" key="two" className="usa-nav__link">
```
[the remark this answers is not in the document]
Comment (hilvitzs) [reply tdrs-frontend/src/components/Header/Header.jsx]: fixed

File: tdrs-frontend/src/components/Header/Header.jsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+} from '@trussworks/react-uswds'
+
+function HeaderComp() {
+  const [mobileNavOpen, setMobileNavOpen] = useState(false)
+
+  const toggleMobileNav = () => {
+    setMobileNavOpen((prevOpen) => !prevOpen)
+  }
+
+  const testItemsMenu = [
```
[the remark this answers is not in the document]
Comment (hilvitzs) [reply tdrs-frontend/src/components/Header/Header.jsx]: fixed

File: tdrs-frontend/src/components/Welcome/Welcome.jsx [hunk 5]
```diff
@@ -40,25 +30,64 @@ function Welcome() {
...
+            </h2>
+            <p>Questions about TANF data?</p>
+            <p>Email: tanfdata@acf.hhs.gov</p>
+          </Grid>
+          <Grid tablet={{ col: 8 }}>
+            <img
+              className="usa-media-block__img"
+              src={circleImg}
+              style={{ height: '120px' }}
+              alt="Alt text"
```
[the remark this answers is not in the document]
Comment (hilvitzs) [reply tdrs-frontend/src/components/Welcome/Welcome.jsx]: these were removed

File: tdrs-frontend/src/components/Welcome/Welcome.jsx (hunk 1 above)
Comment (hilvitzs) [reply tdrs-frontend/src/components/Welcome/Welcome.jsx]: these were removed

**Review by carltonsmith — APPROVED** [review]
These changes look good to me

**Review by RafterGit — APPROVED** [review]

**Pull request merged**